include/envoy/formatter/substitution_formatter.h

```diff
@@ -78,5 +79,47 @@ class FormatterProvider {
 
 using FormatterProviderPtr = std::unique_ptr<FormatterProvider>;
 
+/**
+ * Interface for command parser.
+ * CommandParser returns a FormatterProviderPtr after successfully parsing an access log format
+ * token, nullptr otherwise.
+ */
+class CommandParser {
+public:
+  virtual ~CommandParser() = default;
+
+  /**
+   * Return a FormatterProviderPtr if this command is parsed from the token.
+   * @param token the token to parse
+   * @param pos current position in the entire format string
+   * @param command_end_position position at the end of the command token
```

Comment:
I'm not very familiar with the formatting code, but it's pretty confusing. Can you provide a small example of what token/pos/command_end_position might actually be?

Reply:
added something